File: frontend/src/types/order.ts
export type OrderState =
  | "SIGNAL"
  | "RISK_CHECKED"
  | "ORDER_REQUESTED"
  | "ORDER_SENT"
  | "RECONCILING"
  | "PARTIALLY_FILLED"
  | "FILLED"
  | "CANCEL_REQUESTED"
  | "CANCELED"
  | "REJECTED"
  | "ERROR";

export interface Order {
  id: string;
  symbol: string;
  side: "BUY" | "SELL";
  quantity: number;
  price: number;
  filled_quantity: number;
  average_fill_price: number | null;
  broker_order_id: string | null;
  source: "MANUAL" | "STRATEGY" | "EXIT" | "RECOVERY";
  commission: number;
  tax: number;
  reprice_count: number;
  mode: string;
  state: OrderState;
  message: string;
  created_at: string;
  updated_at: string;
}

export interface ManualOrderInput {
  symbol: string;
  side: "BUY" | "SELL";
  quantity: number;
  price: number;
  live_confirmation?: string;
}

File: backend/app/schemas/order.py
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field

from app.trading.order_state import OrderState


class ManualOrderRequest(BaseModel):
    symbol: str = Field(pattern=r"^\d{6}$", description="국내주식 6자리 종목코드")
    side: Literal["BUY", "SELL"]
    quantity: int = Field(gt=0)
    price: Decimal = Field(gt=0)
    live_confirmation: str | None = None


class OrderResponse(BaseModel):
    order_id: str
    mode: str
    state: OrderState
    message: str
    broker_order_id: str | None = None
    filled_quantity: int = 0
    average_fill_price: Decimal | None = None
    risk_reasons: list[str] = Field(default_factory=list)


class CancelOrderResponse(BaseModel):
    order_id: str
    state: OrderState
    message: str


class KillSwitchRequest(BaseModel):
    stopped: bool = True
